src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,426 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use BadMethodCallException;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\Client;
+use MongoDB\Collection;
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\WriteException;
+use MongoDB\Exception\DriverRuntimeException;
+use MongoDB\Exception\InvalidArgumentException as MongoInvalidArgumentException;
+use MongoDB\Exception\UnsupportedException;
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockExpiredException;
+use Symfony\Component\Lock\Exception\LockStorageException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * MongoDbStore is a StoreInterface implementation using MongoDB as a storage
+ * engine.
+ *
+ * CAUTION: For MongoDB servers ^2.2 TTL Indexes are used so this store relies
+ * on all client and server nodes to have synchronized clocks for lock expiry
+ * to occur at the correct time. For MongoDB servers <2.2 this store relies
+ * on all client nodes to have synchronized clocks for lock expiry to occur at
+ * the correct time. To ensure locks don't expire prematurely; the TTLs should
+ * be set with enough extra time to account for any clock drift between nodes.
+ *
+ * CAUTION: The locked resource name is indexed in the _id field of the
+ * lock collection. An indexed field's value in MongoDB can be a maximum
+ * of 1024 bytes in length inclusive of structural overhead.
+ *
+ * @see https://docs.mongodb.com/manual/reference/limits/#Index-Key-Limit
+ *
+ * @requires extension mongodb
+ *
+ * @author Joe Bennett <joe@assimtech.com>
+ */
+class MongoDbStore implements StoreInterface
+{
+    private $mongo;
+    private $options;
+    private $initialTtl;
+
+    private $collection;
+    private $databaseVersion;
+
+    /**
+     * Returns whether or not the store is supported.
+     *
+     * @return bool
+     *
+     * @internal
+     */
+    public static function isSupported(): bool
```

Comment:
imho this is not needed

Reply:
I did this because some of Jérémy's stores do it. I thought it might be useful later if a bundle extension needs to check support to auto default/configure something supported, but maybe we could add an interface for this later to all stores if we ever did something like that. Also thinking about it more... you could never auto configure a MongoDbStore since the database name option is mandatory. So I have removed `static::isSupported()`.